src/components/Main/Main.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+interface Props {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  processingTodos: number[];
+}
+
+export const Main: React.FC<Props> = ({ todos, tempTodo, processingTodos }) => {
+  return (
+    <section className="todoapp__main">
+      <TransitionGroup>
+        {todos.map(todo => (
+          <CSSTransition
+            key={todo.id}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem
+              {...todo}
+              isLoading={processingTodos.includes(todo.id)}
+            />
+          </CSSTransition>
+        ))}
+
+        {tempTodo && (
+          <CSSTransition
+            key={0}
+            timeout={300}
+            classNames="temp-item"
+          >
+            <TodoItem
+              {...tempTodo}
```

Comment:
it's not good practice to pass props in such a way. It's better to explicitly pass each prop you need